File: 2024/day_06/solution.js
function inBox(box) {
  for (let i = 1; i < box.length - 1; i++) {
    if (
      box[i].includes("*") &&
      box[i].indexOf("*") > 0 &&
      box[i].indexOf("*") < box[i].length - 1
    ) {
      return true;
    }
  }
  return false;
}

console.log(inBox(["###", "#*#", "###"])); // ➞ true
console.log(inBox(["####", "#* #", "#  #", "####"])); // ➞ true
console.log(inBox(["*####", "#   #", "#  #*", "####"])); // ➞ false
console.log(inBox(["#####", "#   #", "#   #", "#   #", "#####"])); // ➞ false

File: 2024/day_06/solution.py
from typing import List

def in_box(box: List[str]) -> bool:
  # Iterate over the rows of the box, excluding the first and last rows
  for row in box[1:-1]:
    if "*" in row and 0 < row.index("*") < len(row) - 1:
      return True
    
  return False
